FHEM/00_SIGNALduino.pm

```diff
@@ -2856,23 +2858,102 @@ function SD_plistWindow(txt)
   $(div).html(txt);
   $("body").append(div);
   var oldPos = $("body").scrollTop();
+  var element = document.getElementById("SD_protoCaption");
+  var caption = element.innerHTML;
+  var btxtStable = "";
+  var btxtBlack = "";
+  if (caption.substr(0,1) != "d") {
+    btxtStable = "stable";
+  }
+  if (caption.substr(-1) == ".") {
+    btxtBlack = " except blacklist";
+  }
   $(div).dialog({
     dialogClass:"no-close", modal:true, width:"auto", closeOnEscape:true, 
     maxWidth:$(window).width()*0.9, maxHeight:$(window).height()*0.9,
-    buttons: [{text:"OK", click:function(){
-      $(this).dialog("close");
-      $(div).remove();
-      location.reload();
-    }}]
+    title: "Protocollist Overview",
+    buttons: [
+      {text:"select all " + btxtStable + btxtBlack, click:function(){
+           //FW_cmd(FW_root+ \'?XHR=1&cmd={SIGNALduino_FW_selectAll()}\', function(data){SD_selectAll(data)});
+      }},
+      {text:"deselect all", click:function(){
+           // FW_cmd(FW_root+ \'?XHR=1&cmd={SIGNALduino_FW_deselectAll()}\', function(){SD_deselectAll()});
+           SD_deselectAll()
+      }},
+      {text:"save to whitelist and close", click:function(){
+          var ret = SD_saveWhitelist();
+          FW_cmd(FW_root+ \'?XHR=1&cmd={SIGNALduino_FW_saveWhitelist("\'+ret+\'")}\');
+          $(this).dialog("close");
+          $(div).remove();
+          location.reload();
+      }},
+      {text:"close", click:function(){
+        $(this).dialog("close");
+        $(div).remove();
+        location.reload();
+      }}]
   });
 }
+
+function SD_deselectAll()
+{
+  var element = document.getElementById("SD_protocolDialog");
+  var checkboxes = element.getElementsByTagName("input");
+  for (var i = 0; i < checkboxes.length; i++) {
+     checkboxes[i].checked = false;
+  }
+}
+
+function SD_selectAll(data)
+{
+
+}
+
+function SD_saveWhitelist()
+{
+  var element = document.getElementById("SD_protocolDialog");
+  var checkboxes = element.getElementsByTagName("input");
+  var txt = "";
+  for (var i = 0; i < checkboxes.length; i++) {
+    if (checkboxes[i].checked == true) {
+      txt = txt + checkboxes[i].name + ",";
+    }
+  }
+  return txt;
+}
+
+
 </script>';
   return $ret;
-  
-  
 }
 
+sub SIGNALduino_FW_saveWhitelist
+{
+	my $wl_attr = shift;
+	#my $hash = shift;
+	my $hash = $glHash;
```

Comment:
was genau soll das mit dem $glHash bringen?

Reply:
Dies ist ein workaround, da es mir nicht gelungen ist den hash zu übergeben.

Reply:
JSON eignet sich recht gut um Daten zu übergeben. 
Hier würde es aber doch problemlos ausreichen einfach den Namen der Definition zu übergeben und dann den Hash zu suchen. So hatte ich es für das Aktivieren / Deaktivieren auch gemacht.

Reply:
Ich habe es so wie Du es gemacht hast auch versucht den $name zu übergeben
`FW_cmd(FW_root+ \'?XHR=1&cmd={SIGNALduino_FW_selectAll("'.$name.'")}\', function(data){SD_selectAll(data)});`
bekomme aber eine Fehlermeldung
`PERL WARNING: Use of uninitialized value $name in concatenation (.) or string`

Reply:
in welcher Funktion hast Du das versucht?

Reply:
in der` function SD_plistWindow(txt)`

Reply:
Damit ich den namen mit übergeben kann, müsste ich in der `SIGNALduino_FW_getProtocolList` den Namen mit in die html Daten reinschreiben und in der function SD_plistWindow wieder raus holen.

Geht es evtl so?
`$ret .="<caption id=\"SD_protoCaption\" name=$name>$devText";`

Reply:
Nein, Du musst es nicht dort rein schreiben. Du könntest es natürlich in ein verstecktes Formularfeld, aber es ist nicht notwendig.

Die Javascript Funktionen werden ja durch die perl sub erstellt. Und beim erstellen, kann $name auch problemlos eingebaut werden.

Reply:
Das ganze wird mir zu kompliziert und aufwendig.
Die `function SD_plistWindow ` steht doch in der `sub SIGNALduino_FW_Detail($@) {`
wie soll ich da den namen einbauen.

Reply:
Wie immer, einfach wenn man weiss wie es geht. Ich habe es schnell eingebaut.

Was mir jetzt aktuell nicht gut gefällt ist, dass ich ein Haken setzen kann, und das Protokoll wird dann doch nicht aktiviert. Beim Nächsten Laden der Liste ist es dann wieder deaktiviert.

Reply:
Mit diesem commit wird das Protokoll bei gesetztem Haken immer aktiviert
https://github.com/RFD-FHEM/RFFHEM/pull/444/commits/575b6a7ac08287226222694d4a65c4935bf52afa